src/cls/rbd/cls_rbd.cc

```diff
@@ -5052,8 +5052,8 @@ int trash_list(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
   CLS_LOG(20, "trash_get_images");
   while (data.size() < max_return) {
     map<string, bufferlist> raw_data;
-    int max_read = std::min<int32_t>(RBD_MAX_KEYS_READ,
-                                     max_return - data.size());
+    max_read = std::min<int32_t>(RBD_MAX_KEYS_READ,
+                                 max_return - data.size());
     int r = cls_cxx_map_get_vals(hctx, last_read, trash::IMAGE_KEY_PREFIX,
```

Comment:
I think you can keep the declaration of `max_read` outside the while loop, without issues.